// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option objc_class_prefix = "HLW";

package RpcModule;

// 请求消息服务开通
service Coordinator {
  // Sends a Register sign to server
  rpc Register (RegisterRequest) returns (RegisterResponse) {}
  rpc Example(ExampleArgs) returns (ExampleReply){}
  rpc TaskDone(stream TaskResultRequest) returns (TaskRecvReply){}
}

// 注册请求消息
message RegisterRequest {
  string ip=1;
  int32 port=2;

  int32 status=3;
  int32 type=4;

  string task=5;
  string taskresule=6;

  int32 TaskIndex=7;
  bool Done=8;   
}

// 注册响应消息
message RegisterResponse {
  bool success = 1;  
}
//样例
message ExampleArgs{
  int32 X=1;
}
message ExampleReply{
  int32 Y=1;
}
//任务完成
message TaskResultRequest{
  string ip=1;
  int32 port=2;
  string task=3;
  int32 task_id=4;
  int32 task_type=5;
}
message TaskRecvReply{
  bool success=1;
}

